staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -159,6 +160,49 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 	}
 }
 
+// As we don't need a high precision here, we keep all watchers timeout within a
+// second in a bucket, and pop up them once at the timeout. To be more specific,
+// if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
+// This is NOT thread-safe.
+type timeBucketWatchers struct {
+	watchersQueue map[int64][]*cacheWatcher
+	startBucketID int64
+	clock         clock.Clock
+}
+
+func newTimeBucketWatchers(clock clock.Clock) *timeBucketWatchers {
+	return &timeBucketWatchers{
+		watchersQueue: make(map[int64][]*cacheWatcher),
+		startBucketID: clock.Now().Unix(),
+		clock:         clock,
+	}
+}
+
+// Adding a watcher to the bucket, if the deadline is before the start, it will be
```

Comment:
nit: the go convention is to start the doc string with the function name, e.g. "addWatcher adds a watcher to the bucket...".  While our linters don't require this format for non exported functions, it's good to follow the convention where we can.

Reply:
done